modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java

```diff
@@ -546,16 +542,19 @@ private JSONArray _getFieldTypesJSONArray(
 			}
 
 			for (JSONObject jsonObject : (Iterable<JSONObject>)jsonArray) {
-				String[] fieldTypeScopes = StringUtil.split(
-					jsonObject.getString("scope"));
+				Boolean match = false;
 
-				boolean anyMatch = Stream.of(
-					fieldTypeScopes
-				).anyMatch(
-					scope -> scopes.contains(scope)
-				);
+				for (String scope :
+						StringUtil.split(jsonObject.getString("scope"))) {
+
+					if (scopes.contains(scope)) {
+						match = true;
 
-				if (anyMatch) {
+						break;
+					}
+				}
+
+				if (match) {
```

Comment:
Hi @dantewang, 
Here `StringUtil.split` method returns an array. 
Using ListUtil, I need to convert it to list  ` Array.asList` first. 
It will create an ArrayList.

Reply:
@JanisZhang It's just a wrapper on top of the array, not the normal `ArrayList`, so it's OK.